cameraGoogleAPI/main.js

```diff
@@ -0,0 +1,141 @@
+const url = require('url');
+const fs = require('fs');
+const { http, https } = require('follow-redirects');
+const CamOverlayAPI = require('camstreamerlib/CamOverlayAPI');
+
+var settings = null;
+var co = null;
+var default_period_time = 1000;
+function run() {
+  try {
+    var data = fs.readFileSync(process.env.PERSISTENT_DATA_PATH + 'settings.json');
+    settings = JSON.parse(data);
+  } catch (err) {
+    console.log('No settings file found');
+    return;
+  }
+
+
+  co = new CamOverlayAPI({
+    'ip': '127.0.0.1',
+    'port': 80,
+    'auth': settings.camera_user + ':' + settings.camera_pass,
+    'serviceID': settings.overlay_id
+  });
+
+  co.on('error', (err) => {
+    console.log('COAPI-Error: ' + err);
+  });
+
+  co.on('close', () => {
+    console.log('COAPI-Error: connection closed');
+    process.exit(1);
+  });
+
+  co.connect().then(async function(){
+    oneAppPeriod(co);
+  }, () => {
+    console.log('COAPI-Error: connection error');
+    process.exit(1);
+  });
+}
+
+function sendRequest(raw_url, auth) {
```

Comment:
Tohle je ok, ale muzes priste mrknout i na tridu v CamStreamerLib HTTPRequest, ktera asi zatim neumi https, ale zas umi resit Digest autorizaci. Je otazka jeslti neni lepsi ji vic pouzivat, ale je primarne myslena pro requesty do kamery, kde prave muze byt nastavena digest autorizace.

Reply:
fixed.
možná by bylo fajn mít tam i funkci HTTPSrequest, aby se dalo snáz prohazovat, když chci použít jiný prokol. mám tam tu funkci dopsat?

Reply:
To necham na tobe..asi bych pripadne neduplikoval cely kod a pridal tam podobnou podminku jako mas s http/https v tomhle kodu.

Reply:
Tak jak jsi to upravil bych rek ze to je horsi nez to bylo..Bud bych pouzil jeden pristup nebo druhej..pro http pozivat jinej nez pro https je zvlasti a spis jsem ti to psal abys o ty funkci vedel..

Reply:
ok